src/Query/Group.ts

```diff
@@ -151,13 +162,44 @@ export class Group {
         return [Group.escapeMarkdownCharacters(filename)];
     }
 
-    private static groupByRoot(task: Task): string[] {
-        const path = task.path.replace(/\\/g, '/');
-        const separatorIndex = path.indexOf('/');
-        if (separatorIndex == -1) {
-            return ['/'];
+    private static groupByFn(task: Task, arg?: GroupingArg): string[] {
+        const paramsArgs: [string, any][] = [
+            ['description', task.description],
+            ['done', task.doneDate],
+            ['due', task.dueDate],
+            ['filename', task.filename],
+            ['happens', new HappensDateField().earliestDate(task)],
+            ['header', task.precedingHeader],
+            ['markdown', task.originalMarkdown],
+            ['path', task.path.replace('.md', '')],
+            ['priority', task.priority],
+            ['recurrence', task.recurrence],
+            ['root', Group.root(task)],
+            ['scheduled', task.scheduledDate],
```

Comment:
Please could tests include a use of each of those fields in the group, to show how they are rendered?
I am interested in seeing how dates are rendered, for example.